src/ResourceManager/ManagedServiceIdentity/Commands.ManagedServiceIdentity/UserAssignedIdentities/RemoveAzureRmUserAssignedIdentityCmdlet.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System.Globalization;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.ManagedServiceIdentity.Common;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+
+namespace Microsoft.Azure.Commands.ManagedServiceIdentity.UserAssignedIdentities
+{
+    [Cmdlet(VerbsCommon.Remove, "AzureRmUserAssignedIdentity", SupportsShouldProcess = true)]
+    [OutputType(typeof(bool))]
+    public class RemoveAzureRmUserAssignedIdentityCmdlet : MsiBaseCmdlet
+    {
```

Comment:
Need to be able to add piping scenarios for: Get-AzureRmUserAssignedIdentity | Remove-AzureRmUserAssignedIdentity and Find-AzureRmResource | Remove-AzureRmUserAssignedIdentity.

https://github.com/Azure/azure-powershell/wiki/Piping-in-PowerShell#azure-powershell-piping-scenarios

This involves adding two new parameter sets with parameters "InputObject" that accepts a PSUserAssignedIdentity and "ResourceId" which accept and parsed the ResourceId of a UserAssignedIdentity.

Reply:
I have added the InputObject and ResourceId parameter sets. Get-AzureRmUserAssignedIdentity | Remove-AzureRmUserAssignedIdentity works.

But Get-AzureRmResource -ResourceGroupName xxx -Name yyy | Remove-AzureRmUserAssignedIdentity does not resolve to a ParameterSet. Can you guide if anything obvious seems missing?

Same with ResourceId parameter set.

Reply:
@varunkch If you remove the ValueFromPipelineByPropertyName tag from Name and ResourceGroupName it should solve this issue (the issue is that get-azurermresource will return an object with properties resourcegroupname, resourceid, and name, so PowerShell cannot resolve the parameter set.  We normally would want ResourceGroupName and Name to have ValueFromPipelineByPropertyName, but since we have the InputObject parameter, that is no longer necessary).